packages/dd-trace/src/plugins/util/web.js

```diff
@@ -361,6 +364,32 @@ const web = {
     const res = context.res
 
     scope.bind(res, context.span)
+  },
+  obfuscateQs (obfuscator, url) {
+    if (obfuscator === false) return url
+
+    const i = url.indexOf('?')
+
+    if (i === -1) return url
+
+    const path = url.slice(0, i)
+
+    if (obfuscator === true) return path
+
+    let qs = url.slice(i + 1)
+
+    try {
+      qs = vm.runInNewContext('decodeURIComponent(qs).replace(obfuscator, \'<redacted>\')', {
+        qs,
+        obfuscator
+      }, {
+        timeout: 200
+      })
+
+      return `${path}?${qs}`
+    } catch {
```

Comment:
Any way to avoid ever throwing since this is a hot path? Errors could result in an application starving for resources and ultimately crashing.

Reply:
As far as I can tell, the only way to protect against catastrophic regexes without a native addon is the vm timeout, that throws when a timeout occurs, and I don't see any option to change that behavior.
In any case, it should only throw when the timeout is reached ie: the app is under a regex backtracking attack, so the ressources taken by that exception cannot ever be worse than the app literally just freezing because of 1 singular payload crafted for backtracking.

Reply:
Well it just protects against the situation where the user sets an unsafe regex and freeze his own app with a legit payload. Not against an attack. I don't see how we could protect against an attack. 
A lower timeout is probably fine I guess, but currently that's the value specified in the RFC. 
What would a better concurrency logic look like to you ?